File: frontend/types/MajorRequest.ts
export interface MajorRequest {
    id?: number;
    programId: number;
    nameTh: string;
    nameEn: string;
    tuitionFees: number;
    language: string;
    courses: File | undefined;
    image: File | undefined;
    description: string;
    careers: string;
    status: boolean;
}

File: backend/src/main/java/com/pbru/it/backend/DTO/request/MajorRequest.java
package com.pbru.it.backend.DTO.request;

import java.math.BigDecimal;

import org.springframework.web.multipart.MultipartFile;

import com.pbru.it.backend.Models.CourseLanguage;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MajorRequest {

    private Long programId;

    private String nameTh;

    private String nameEn;

    private BigDecimal tuitionFees;

    private CourseLanguage language;

    private MultipartFile courseFile;

    private MultipartFile image;

    private String description;

    private String careers;

    private Boolean status;

}
